data/helpers.d/backend

```diff
@@ -213,3 +213,209 @@ ynh_remove_fpm_config () {
+#        If possible, prefer to use major version number (e.g. 8 instead of 8.10.0).
+#        The crontab will handle the update of minor versions when needed.
+ynh_install_nodejs () {
+	# Use n, https://github.com/tj/n to manage the nodejs versions
+	nodejs_version="$1"
+	local n_install_script="https://git.io/n-install"
+
+	# Create $n_install_dir
+	mkdir -p "$n_install_dir"
+
+	# Load n path in PATH
+	CLEAR_PATH="$n_install_dir/bin:$PATH"
+	# Remove /usr/local/bin in PATH in case of node prior installation
+	PATH=$(echo $CLEAR_PATH | sed 's@/usr/local/bin:@@')
+
+	# Move an existing node binary, to avoid to block n.
+	test -x /usr/bin/node && mv /usr/bin/node /usr/bin/node_n
+	test -x /usr/bin/npm && mv /usr/bin/npm /usr/bin/npm_n
+
+	# If n is not previously setup, install it
+	if ! test n --version > /dev/null 2>&1
+	then
+		ynh_install_n
+	fi
+
+	# Modify the default N_PREFIX in n script
+	ynh_replace_string "^N_PREFIX=\${N_PREFIX-.*}$" "N_PREFIX=\${N_PREFIX-$N_PREFIX}" "$n_install_dir/bin/n"
+
+	# Restore /usr/local/bin in PATH
+	PATH=$CLEAR_PATH
+
+	# And replace the old node binary.
+	test -x /usr/bin/node_n && mv /usr/bin/node_n /usr/bin/node
+	test -x /usr/bin/npm_n && mv /usr/bin/npm_n /usr/bin/npm
+
+	# Install the requested version of nodejs
+	n $nodejs_version
+
+	# Find the last "real" version for this major version of node.
+	real_nodejs_version=$(find $node_version_path/$nodejs_version* -maxdepth 0 | sort --version-sort | tail --lines=1)
+	real_nodejs_version=$(basename $real_nodejs_version)
+
+	# Create a symbolic link for this major version if the file doesn't already exist
+	if [ ! -e "$node_version_path/$nodejs_version" ]
+	then
+		ln --symbolic --force --no-target-directory $node_version_path/$real_nodejs_version $node_version_path/$nodejs_version
+	fi
+
+	# Store the ID of this app and the version of node requested for it
+	echo "$YNH_APP_ID:$nodejs_version" | tee --append "$n_install_dir/ynh_app_version"
+
+	# Store nodejs_version into the config of this app
+	ynh_app_setting_set $app nodejs_version $nodejs_version
+
+	# Build the update script and set the cronjob
+	ynh_cron_upgrade_node
+
+	ynh_use_nodejs
+}
+
+# Remove the version of node used by the app.
+#
+# This helper will check if another app uses the same version of node,
+# if not, this version of node will be removed.
+# If no other app uses node, n will be also removed.
+#
+# usage: ynh_remove_nodejs
+ynh_remove_nodejs () {
+	ynh_use_nodejs
+
+	# Remove the line for this app
+	sed --in-place "/$YNH_APP_ID:$nodejs_version/d" "$n_install_dir/ynh_app_version"
+
+	# If no other app uses this version of nodejs, remove it.
+	if ! grep --quiet "$nodejs_version" "$n_install_dir/ynh_app_version"
+	then
+		n rm $nodejs_version
+	fi
+
+	# If no other app uses n, remove n
+	if [ ! -s "$n_install_dir/ynh_app_version" ]
+	then
+		ynh_secure_remove "$n_install_dir"
+		ynh_secure_remove "/usr/local/n"
+		sed --in-place "/N_PREFIX/d" /root/.bashrc
```

Comment:
What case is this supposed to cover?

Reply:
Unless it's also an old useless line, 'n' will add a line to the .bashrc, to load n.
Maybe, we could remove this line just after the installation.

Reply:
OK, that would require at least a comment for this :wink:

Reply:
Let me check if it's still relevant before.
And also see if we can remove it just after the installation.

Reply:
So... It's an old useless line...
This helper has been refactored many times...